modulus/models/gnn_layers/distributed_graph.py

```diff
@@ -769,6 +908,12 @@ def get_src_node_features_in_partition(
     ) -> torch.Tensor:  # pragma: no cover
         # if global features only on local rank 0 also scatter, split them
         # according to the partition and scatter them to other ranks
+
+        if self.graph_partition.matrix_decomp:
```

Comment:
Ideally, one shouldn't make the code using these utilities dependent on how a graph is partitioned. Couldn't one instead of throwing this error just use `get_dst_node_features` underneath?

Reply:
Thanks @stadlmax and sorry for the late reply... Updated as suggested.